### PR TITLE
Fix handling of unsigned integer types in binops

### LINKED ISSUE #2152: [BUG] Runtime Error while comparing int with bool
**Describe the bug**
We  seem to be getting exception while trying to compare an int with a Boolean series, but we are able to achieve the inverse, i.e., comparing a Boolean with integer Series. But pandas treats all integers as False except 1 as True and compares the equality check.

**Steps/Code to reproduce bug**

```python
import cudf
x = cudf.Series([True, False, False])
print(x == 1)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
cudf/bindings/cudf_cpp.pyx in cudf.bindings.cudf_cpp.set_scalar_value()

ValueError: Cannot convert numpy scalar of dtype uint8to gdf_scalar
Exception ignored in: 'cudf.bindings.cudf_cpp.gdf_scalar_from_scalar'
Traceback (most recent call last):
  File "cudf/bindings/cudf_cpp.pyx", line 168, in cudf.bindings.cudf_cpp.set_scalar_value
ValueError: Cannot convert numpy scalar of dtype uint8to gdf_scalar
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-18-49bd823eaefe> in <module>
      2 import cudf
      3 x = cudf.Series([True, False, False])
----> 4 print(x == 1)

/conda/envs/cudf/lib/python3.7/site-packages/cudf/dataframe/series.py in __eq__(self, other)
    832 
    833     def __eq__(self, other):
--> 834         return self._unordered_compare(other, 'eq')
    835 
    836     def equals(self, other):

/conda/envs/cudf/lib/python3.7/site-packages/cudf/dataframe/series.py in _unordered_compare(self, other, cmpops)
    803         nvtx_range_push("CUDF_UNORDERED_COMP", "orange")
    804         other = self._normalize_binop_value(other)
--> 805         outcol = self._column.unordered_compare(cmpops, other)
    806         result = self._copy_construct(data=outcol)
    807         result.name = None

/conda/envs/cudf/lib/python3.7/site-packages/cudf/dataframe/numerical.py in unordered_compare(self, cmpop, rhs)
     92 
     93     def unordered_compare(self, cmpop, rhs):
---> 94         return numeric_column_compare(self, rhs, op=cmpop)
     95 
     96     def ordered_compare(self, cmpop, rhs):

/conda/envs/cudf/lib/python3.7/site-packages/cudf/dataframe/numerical.py in numeric_column_compare(lhs, rhs, op)
    417 
    418 def numeric_column_compare(lhs, rhs, op):
--> 419     return numeric_column_binop(lhs, rhs, op, out_dtype=np.bool_)
    420 
    421 

/conda/envs/cudf/lib/python3.7/site-packages/cudf/dataframe/numerical.py in numeric_column_binop(lhs, rhs, op, out_dtype, reflect)
    402     out = columnops.column_empty(row_count, dtype=out_dtype, masked=masked)
    403     # Call and fix null_count
--> 404     null_count = cpp_binops.apply_op(lhs, rhs, out, op)
    405 
    406     out = out.replace(null_count=null_count)

cudf/bindings/binops.pyx in cudf.bindings.binops.apply_op()

cudf/bindings/binops.pyx in cudf.bindings.binops.apply_op_v_s()

RuntimeError: cuDF failure at: /cudf/cpp/src/binaryop/binaryop.cpp:200: Input pointers are null
```

Pandas:

```python
import pandas as pd
x = pd.Series([True, False, False])
print(x == 1)
0     True
1    False
2    False
dtype: bool
```
**Expected behavior**
We should also be having the same behavior by treating 1 as `True` and any other value as `False`.

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: from source

**Environment details**
output of the `cudf/print_env.sh` script here:  [env.txt](https://github.com/rapidsai/cudf/files/3343312/env.txt)

### PR DESCRIPTION
**Pull request by shwina**
Fixes #2152

### CODE REVIEW

File: python/cudf/dataframe/numerical.py [hunk 1]
```diff
@@ -105,7 +105,8 @@ def normalize_binop_value(self, other):
         other_dtype = np.min_scalar_type(other)
         if other_dtype.kind in 'biuf':
             other_dtype = np.promote_types(self.dtype, other_dtype)
-
+            if other_dtype.kind in 'u':
+                other_dtype = np.dtype(other_dtype.name.lstrip('u'))
```
[the remark this answers is not in the document]
Comment (kkraus14) [reply python/cudf/dataframe/numerical.py]: We have `cudf.utils.utils.min_scalar_type(...)` but it has the same problem as this function.

[the remark this answers is not in the document]
Comment (shwina) [reply python/cudf/dataframe/numerical.py]: How about something along these lines:

```
other_dtype = np.min_scalar_type(-np.iinfo(other_dtype).max)
```

[the remark this answers is not in the document]
Comment (kkraus14) [reply python/cudf/dataframe/numerical.py]: @shwina that will unnecessarily promote int8 --> int16, and int16 --> int32, for positive numbers that would fit in it.

File: python/cudf/utils/utils.py [hunk 2]
```diff
@@ -242,3 +242,13 @@ def min_scalar_type(a, min_size=8):
...
+
+
+def min_signed_type(x):
+    """
+    Return the smallest *signed* integer dtype
+    that can represent the integer ``x``
+    """
+    for int_dtype in np.sctypes['int']:
+        if np.iinfo(int_dtype).min <= x <= np.iinfo(int_dtype).max:
+            return int_dtype
```
Comment (kkraus14) [inline python/cudf/utils/utils.py]: Can we update the above `min_scalar_type` to use this unless/until we support unsigned types?

Comment (shwina) [reply python/cudf/utils/utils.py]: Done

**Comment by kkraus14** [comment]
@shwina this just needs the merge conflicts resolved and it will be good to go.

**Pull request merged**